tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp

```diff
@@ -360,3 +387,269 @@ TEST_F(CRTLoggingTest, testTraceLogLevel)
+                                    {
+                                        LogOnCRTLogSystemInterfaceWithoutVaArgs(logSystem.get(), LogLevel::Info, logMsgTag.c_str(), "All work and no play makes Jack a dull boy");
+                                    }
+
+                                    return true;
+                                });
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    run = true;
+
+    for(size_t i = 0; i < PARALLEL_COUNT; ++i)
+    {
+        bool success = futures[i].get();
+        ASSERT_TRUE(success);
+    }
+}
+
+TEST_F(LoggingTestLogFileRace, testCrtLoggerRaceMultipleLoggers)
+{
+    /*
+     * Tests a race condition on the DefaultCRTLogSystem, create CRT logger per each thread.
+     * It is a test on an SDK component, not on a test helper.
+     */
+    static const size_t PARALLEL_COUNT = 32;
+    Aws::Vector<std::future<bool>> futures(PARALLEL_COUNT);
+
+    std::atomic<bool> run;
+    run = false;
+
+    for(size_t i = 0; i < PARALLEL_COUNT; ++i)
+    {
+        futures[i] = std::async(std::launch::async,
+                                [i, &run]()-> bool
+                                {
+                                    while(true)
+                                    {
+                                        if(run.load())
+                                        {
+                                            break;
+                                        }
+                                    }
+
+                                    Aws::UniquePtr<DefaultCRTLogSystem> logSystem =
+                                            Aws::MakeUnique<DefaultCRTLogSystem>(AllocationTag, LogLevel::Info);
+                                    Aws::String logMsgTag = "testCrtRaceOnLogFile thread #" + Aws::Utils::StringUtils::to_string(i);
+                                    LogOnCRTLogSystemInterfaceWithoutVaArgs(logSystem.get(), LogLevel::Info, logMsgTag.c_str(), "Knock knock");
+                                    LogOnCRTLogSystemInterfaceWithoutVaArgs(logSystem.get(), LogLevel::Info, logMsgTag.c_str(), " - Who's there?");
+                                    LogOnCRTLogSystemInterfaceWithoutVaArgs(logSystem.get(), LogLevel::Info, logMsgTag.c_str(), " - Race condition!");
+                                    for(size_t j = 0; j < 100; ++j)
+                                    {
+                                        LogOnCRTLogSystemInterfaceWithoutVaArgs(logSystem.get(), LogLevel::Info, logMsgTag.c_str(), "All work and no play makes Jack a dull boy");
+                                    }
+
+                                    return true;
+                                });
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    run = true;
+
+    for(size_t i = 0; i < PARALLEL_COUNT; ++i)
+    {
+        bool success = futures[i].get();
+        ASSERT_TRUE(success);
+    }
+}
+
+TEST_F(LoggingTestLogFileRace, testCrtLoggerRaceSingleLogger)
```

Comment:
we could maybe create a common test function that takes a `std::function` and pass the logging lambda to it instead of duplicating setup/tear down logic

Reply:
That's a great suggestion, but it is a test code, I'll better switch to next task... I can revisit it later.